plugin-server/src/worker/vm/extensions/posthog.ts

```diff
@@ -41,6 +42,7 @@ async function queueEvent(hub: Hub, pluginConfig: PluginConfig, data: InternalDa
                     now: data.timestamp,
                     sent_at: data.timestamp,
                     uuid: data.uuid,
+                    ...(data.groups ? { groups: data.groups } : {}),
```

Comment:
The key is incorrect - see https://posthog.com/docs/user-guides/group-analytics for other libraries, the key is $groups not groups.